cpp/util/Histogram.h

```diff
@@ -250,6 +250,81 @@ template<typename T> class Histogram
             m_local_histograms.local().increment(value_bin, weight);
         }
 
+        void reduceHistogramsInto(ManagedArray<T>& result)
+        {
+            typedef std::vector<util::Histogram<T>*> HistogramVector;
+            const unsigned int BODY_DEFAULT(0xffffffff);
+            struct reduceThreadLocalHistogram
+            {
+                const HistogramVector& m_;
+                const std::vector<size_t> shape_;
```

Comment:
You only need `size_`, so I'd remove `shape_`. Do the same for the `ThreadStorage` version.

Reply:
Good catch -- originally I needed `shape_` but not anymore.